xonsh/environ.py

```diff
@@ -201,13 +206,32 @@ def wrapper(*args, **kwargs):
         return func(*args, **kwargs)
     return wrapper
 
+def ensure_hg(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        kwargs['cwd'] = kwargs.get('cwd', _get_cwd())
+        if kwargs['cwd'] is None:
+            return
+
+        # bail if hglib is not installed
+        if hglib is None:
+            return
+
+        try:
+            c = hglib.open(kwargs['cwd'])
+            c.tip()
+        except OSError:
+            # hglib is installed but hg isn't, so bail
+            return
+        except hglib.error.CommandError:
```

Comment:
I didn't see that on mercurial 3.4, I'll try testing it out with a build of mercurial 2.2.